src/test/java/org/jenkinsci/pipeline_steps_doc_generator/PipelineStepExtractor.java

```diff
@@ -0,0 +1,193 @@
+package org.jenkinsci.pipeline_steps_doc_generator;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+import org.jenkinsci.plugins.structs.describable.DescribableModel;
+import org.jenkinsci.plugins.structs.describable.DescribableParameter;
+import org.jenkinsci.plugins.structs.describable.HeterogeneousObjectType;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jvnet.hudson.reactor.Reactor;
+import org.jvnet.hudson.reactor.ReactorException;
+import org.jvnet.hudson.reactor.Task;
+import org.jvnet.hudson.reactor.TaskBuilder;
+import org.jenkinsci.infra.tools.HyperLocalPluginManager;
+
+import hudson.MockJenkins;
+import hudson.init.InitMilestone;
+import hudson.init.InitStrategy;
+import hudson.model.Describable;
+import hudson.model.Descriptor;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import jenkins.InitReactorRunner;
+import jenkins.model.Jenkins;
+
+/**
+ * Process and find all the Pipeline steps definied in Jenkins plugins.
+ */
+public class PipelineStepExtractor {
```

Comment:
I do also want to point out that having a separate class for setting up the HyperLocalPluginManager and running the reactor.  This is a great idea!